torch/csrc/jit/passes/onnx/function_substitution.cpp

```diff
@@ -60,28 +133,62 @@ void functionCallSubstitution(Block* block) {
         const std::string& name = cur->s(attr::name);
         if (auto class_type = cur->input(0)->type()->cast<ClassType>()) {
           Function& function = class_type->getMethod(name);
+          ScopePtr call_scope = ForwardCallScope(*graph, cur);
+          WithCurrentScope scope_guard(*graph, call_scope);
+          GRAPH_DEBUG(
+              "Setting scope guard for forward call: ",
+              graph->current_scope()->namesFromRoot());
           if (auto graphFunction = tryToGraphFunction(function)) {
+            GRAPH_DEBUG(
+                "Inner graph for method call ",
+                name,
+                ": ",
+                *graphFunction->graph());
+            WithCurrentScope inner_graph_scope_guard(
+                *graphFunction->graph(), call_scope);
             functionCallSubstitution(graphFunction->graph()->block());
             inlineCallTo(cur, graphFunction, false);
           }
         }
       } break;
       default: {
+        if (!graph->current_scope()->isBlank()) {
+          cur->setScope(graph->current_scope());
+        }
         for (auto b : cur->blocks()) {
           functionCallSubstitution(b);
         }
       } break;
     }
+    GRAPH_DEBUG(
+        "Graph current scope after node process: ",
+        graph->current_scope()->namesFromRoot());
+  }
+}
+
+ScopePtr ONNXGraphTopLevelScope(Graph& graph) {
+  if (graph.inputs().size() == 0) {
+    return graph.current_scope();
+  }
+  if (auto top_module_type = graph.inputs().at(0)->type()->cast<ClassType>()) {
+    auto scope_name = ::torch::jit::onnx::ONNXScopeName::CreateFullScopeName(
```

Comment:
note: (repeated from another PR) I realized the pytorch project conventionally uses camelCase for function names (instead of PascalCase)